Paintroid/src/main/java/org/catrobat/paintroid/tools/implementation/TextTool.kt

```diff
@@ -218,11 +255,43 @@ class TextTool(
         val maxTextWidth = multilineText.maxOf { line ->
             textPaint.measureText(line)
         }
-
         boxHeight = textHeight * multilineText.size + 2 * BOX_OFFSET
         boxWidth = maxTextWidth + 2 * BOX_OFFSET
     }
 
+    private fun storeAttributes() {
+        oldBoxWidth = boxWidth
+        oldBoxHeight = boxHeight
+        oldToolPosition = PointF(toolPosition.x, toolPosition.y)
+    }
+
+    private fun storeAttributesWithItalic() {
+        boxWidth *= ITALIC_FONT_BOX_ADJUSTMENT
+        oldBoxWidth = boxWidth
+        oldBoxHeight = boxHeight
+        oldToolPosition = PointF(toolPosition.x, toolPosition.y)
+    }
+
+    private fun applyAttributesWithItalic() {
+        boxWidth = oldBoxWidth / ITALIC_FONT_BOX_ADJUSTMENT
+        boxHeight = oldBoxHeight
+        if (oldToolPosition != null) {
+            toolPosition = oldToolPosition as PointF
+        } else {
+            resetBoxPosition()
+        }
+    }
+
+    private fun applyAttributes() {
+        boxWidth = oldBoxWidth
+        boxHeight = oldBoxHeight
+        if (oldToolPosition != null) {
+            toolPosition = oldToolPosition as PointF
+        } else {
+            resetBoxPosition()
+        }
+    }
```

Comment:
```suggestion
    private fun storeAttributes(italic: Boolean = false) {
        if (italic) {
            boxWidth *= ITALIC_FONT_BOX_ADJUSTMENT
        }
        oldBoxWidth = boxWidth
        oldBoxHeight = boxHeight
        oldToolPosition = PointF(toolPosition.x, toolPosition.y)
    }

    private fun applyAttributes(italic: Boolean = false) {
        boxWidth = oldBoxWidth / if (italic) ITALIC_FONT_BOX_ADJUSTMENT else 1f
        boxHeight = oldBoxHeight
        if (oldToolPosition != null) {
            toolPosition = oldToolPosition as PointF
        } else {
            resetBoxPosition()
        }
    }
```
You can write it like this, to remove duplicity and clean up the code.